modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp

```diff
@@ -112,15 +177,105 @@ void ISO15118_chargerImpl::ready() {
     };
     const auto callbacks = create_callbacks();
 
+    if (mod->config.supported_dynamic_mode) {
+        setup_config.control_mobility_modes.push_back(
+            {iso15118::message_20::ControlMode::Dynamic, iso15118::message_20::MobilityNeedsMode::ProvidedByEvcc});
+        if (mod->config.supported_mobility_needs_mode_provided_by_secc) {
+            setup_config.control_mobility_modes.push_back(
+                {iso15118::message_20::ControlMode::Dynamic, iso15118::message_20::MobilityNeedsMode::ProvidedBySecc});
+        }
+    }
+
+    if (mod->config.supported_scheduled_mode) {
+        setup_config.control_mobility_modes.push_back(
+            {iso15118::message_20::ControlMode::Scheduled, iso15118::message_20::MobilityNeedsMode::ProvidedByEvcc});
+    }
+
+    if (setup_config.control_mobility_modes.empty()) {
+        EVLOG_warning << "Control mobility modes are empty! Setting dynamic mode as default!";
+        setup_config.control_mobility_modes.push_back(
+            {iso15118::message_20::ControlMode::Dynamic, iso15118::message_20::MobilityNeedsMode::ProvidedByEvcc});
+    }
+
     controller = std::make_unique<iso15118::TbdController>(tbd_config, callbacks, setup_config);
-    controller->loop();
+
+    try {
+        controller->loop();
+    } catch (const std::exception& e) {
+        EVLOG_error << e.what();
+    }
 }
 
 iso15118::session::feedback::Callbacks ISO15118_chargerImpl::create_callbacks() {
+
```

Comment:
adding a `using namespace iso15118::message_20` and `using feedback =  iso15118::session::feedback` might be useful here